sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -54,13 +54,23 @@
   /** Returns a {@code WindowedValue} with the given value, timestamp, and windows. */
   public static <T> WindowedValue<T> of(
       T value, Instant timestamp, Collection<? extends BoundedWindow> windows, PaneInfo pane) {
+    return of(value, timestamp, windows, pane, false);
+  }
+
+  /** Returns a {@code WindowedValue} with the given value, timestamp, windows and isRetraction. */
+  public static <T> WindowedValue<T> of(
+      T value,
+      Instant timestamp,
+      Collection<? extends BoundedWindow> windows,
+      PaneInfo pane,
+      boolean isRetraction) {
```

Comment:
Actually I think this is part of the `PaneInfo`. These are encoded efficiently. See `FullWindowedValueCoder`.